### ISSUE TITLE
feat(tsconfig): recommend to enable some options

### ISSUE THREAD

**Issue by zanminkian** [zanminkian/git-validator#88]
- [x] [allowArbitraryExtensions](https://www.typescriptlang.org/tsconfig#allowArbitraryExtensions)
- [ ] ~~[composite](https://www.typescriptlang.org/tsconfig#composite)~~
- [x] [inlineSources](https://www.typescriptlang.org/tsconfig#inlineSources)
- [ ] ~~[noEmitOnError](https://www.typescriptlang.org/tsconfig#noEmitOnError)~~
- [ ] ~~[preserveSymlinks](https://www.typescriptlang.org/tsconfig#preserveSymlinks)~~
- [ ] ~~[importHelpers](https://www.typescriptlang.org/tsconfig#importHelpers)~~
- [x] [moduleDetection](https://www.typescriptlang.org/tsconfig#moduleDetection)

**Comment by zanminkian** [comment]
`preserveSymlinks` is the same flag in Node. In most projects (99.9%), people do not use this flag in Node.

**Comment by zanminkian** [comment]
Files containing `export` or `import` syntax will be treat as `module`. Other files will be treat as `script` file. Every file should be treat as `module`, unless we use `outFile` option and the `module` option is `AMD`/`System` in `tsconfig.json`. Obviously, we don't use `outFile` and we don't set `module` to be `AMD`/`System`. So setting `[moduleDetection](https://www.typescriptlang.org/tsconfig#moduleDetection)` to be `force` makes sense.

Referer:

- https://www.typescriptlang.org/docs/handbook/2/modules.html

**Comment by zanminkian** [comment]
In modern js/ts project, we usually set `module` to be `nodenext`. That means we should try our best to get rid of polyfill. So, we don't need `importHelper`.

**Comment by zanminkian** [comment]
`composite` works for [project references](https://www.typescriptlang.org/docs/handbook/project-references.html). We use should use pnpm or any other monorepo tool to manage each app in the monorepo. Therefore, we don't need `composite`, because we don't use `project references`. Typescript should only do type checking and focus on **one** project.

- If we enable `composite`, people have to specify `rootDir` in their `tsconfig` file, otherwise, the project structure may be out of expected. 
- And there will be a `.tsbuildinfo` file in the project root, which makes project more messy. 
- Enabling `composite` will make it impossible  to fully compile. Every compilation is incremental. 

To sum up, enabling `composite` will bring some trouble to us while we gain nothing from it.
